connector_acp/models/etd_mixin.py

```diff
@@ -60,58 +60,70 @@ def prepare_keywords(self):
     def get_etd_filename(self):
         return self.name or self.number
 
-    def build_xml(self):
+    def build_file(self, file):
         """
-        Build the XML of the record using the company documents and the
-        related XML template and XSD
+        Build the file of the record using the company documents and the
+        related template and
         :return: XML string of the record
         """
         self.set_jinja_env()
-        etd_id = self.get_etd_document()
         # Get the template
-        template = self._env.from_string(base64.b64decode(etd_id.xml).
+        template = self._env.from_string(base64.b64decode(file.template).
                                          decode('utf-8'))
         # Additional keywords used in the template
         kwargs = self.prepare_keywords()
         kwargs.update({
             'o': self,
             'now': fields.datetime.now(),
             'today': fields.datetime.today()})
-        xml_filename = self.get_etd_filename()
-        # Render the XML
-        xml_file = self.File_details(xml_filename + '.xml',
+        filename = self.get_etd_filename()
+        # Render the file
+        res_file = self.File_details(filename + '.' + file.file_type,
                                      template.render(kwargs))
-        xml = str.encode(xml_file.filecontent)
-        # Attache XML file to the document
+        res_content = str.encode(res_file.filecontent)
+        # Attache file to the document
         self.env['ir.attachment'].create({
```

Comment:
Make this optional. Add a `auto_attach_file` boolean to be able to configure this.